rapids-4-spark-tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Qualification.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.tool.profiling
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.rapids.tool.profiling._
+
+/**
+ * Qualification qualifies or disqualifies an application
+ * for GPU acceleration.
+ */
+class Qualification(
+    apps: ArrayBuffer[ApplicationInfo],
+    sqlAggMetricsDF: DataFrame) extends Logging {
+
+  require(apps.nonEmpty)
+  require(!sqlAggMetricsDF.isEmpty)
+  private val fileWriter = apps.head.fileWriter
+
+  // Register sqlAggMetricsDF as a temp view
+  sqlAggMetricsDF.createOrReplaceTempView("sqlAggMetricsDF")
+
+  // Qualify each App
+  for (app <- apps) {
+    qualifyApp(app)
+  }
+
+  //function to qualify an application
```

Comment:
add more description as to what qualify is

Reply:
Added more description. Please let me know if it's okay.